pre_commit/languages/docker.py

```diff
@@ -8,13 +10,43 @@
 from pre_commit.languages import helpers
 from pre_commit.prefix import Prefix
 from pre_commit.util import clean_path_on_failure
+from pre_commit.util import cmd_output_b
 
 ENVIRONMENT_DIR = 'docker'
 PRE_COMMIT_LABEL = 'PRE_COMMIT'
 get_default_version = helpers.basic_get_default_version
 healthy = helpers.basic_healthy
 
 
+def _is_in_docker() -> bool:
+    try:
+        with open('/proc/1/cgroup', 'rb') as f:
+            return b'docker' in f.read()
+    except FileNotFoundError:
+        return False
+
+
+def _get_docker_path(path: str) -> str:
+    if not _is_in_docker():
+        return path
+    hostname = socket.gethostname()
+
+    _, out, _ = cmd_output_b(
+        ' '.join(['docker', 'inspect', hostname]),
+    )
+
+    docker_output = json.loads(out)
+    for mount_path in docker_output[0]['HostConfig']['Binds']:
+        src_path, to_path, *_ = mount_path.split(':')
+        if os.path.commonpath([path, to_path]) != os.path.sep:
```

Comment:
(1) [prefer tuples to lists](https://github.com/asottile/scratch/wiki/Prefer-tuples-to-lists) (2) this isn't quite correct -- you'll need to use `== to_path` I believe:

```pycon
>>> os.path.commonpath(('/a/b/c', '/a/d/b'))
'/a'
```

Reply:
this leads to different behavior on Windows and Linux, so I've updated the source for mounts, added Win-specific test, done.